noether_tpp/src/tool_path_modifiers/organization_modifiers.cpp

```diff
@@ -57,4 +59,175 @@ ToolPaths SnakeOrganizationModifier::modify(ToolPaths tool_paths) const
   return tool_paths;
 }
 
+/**
+ * @brief Organizes the tool path segments within each tool path in the container such that the beginning of a tool path
+ * is as close as possible to the end of the previous tool path (or a reference position, in the case of the first tool
+ * path)
+ * @details The primary intent of this function is to organize tool path segments for edge tool paths; these segments
+ * represent segments of an edge and should be arranged in a sequential and connected order.
+ */
+static std::vector<std::size_t> sortToolPathBySegmentConnection(const ToolPath& path,
+                                                                const Eigen::Vector3d& start_reference)
+{
+  std::vector<std::size_t> segment_order;
+  segment_order.reserve(path.size());
+
+  // Comparator function for finding the segment whose first waypoint is closest to the reference position
+  auto compare_distances = [&path](const std::size_t& a, const std::size_t& b, const Eigen::Vector3d& ref) {
+    double d_a = (path.at(a).at(0).translation() - ref).norm();
+    double d_b = (path.at(b).at(0).translation() - ref).norm();
+    return d_a < d_b;
+  };
+
+  while (segment_order.size() < path.size())
+  {
+    // Get the indices of the yet unordered segments
+    std::vector<std::size_t> remaining_segment_indices;
+    {
+      std::vector<std::size_t> order(path.size());
+      std::iota(order.begin(), order.end(), 0);
+      std::set_difference(order.begin(),
+                          order.end(),
+                          segment_order.begin(),
+                          segment_order.end(),
+                          std::back_inserter(remaining_segment_indices));
+    }
+
+    if (remaining_segment_indices.size() == 1)
+    {
+      segment_order.push_back(remaining_segment_indices.front());
+    }
+    else
+    {
+      // Get the reference point with which to compare the segment start positions
+      Eigen::Vector3d ref;
+      if (segment_order.empty())
+      {
+        // Use the reference start position if no segments have been ordered yet
+        ref = start_reference;
+      }
+      else
+      {
+        // Otherwise use the position of the last waypoint in the last ordered segment
+        ref = path.at(segment_order.back()).back().translation();
+      }
+
+      // Find the segment whose first waypoint is closest to the reference position
+      auto comp = std::bind(compare_distances, std::placeholders::_1, std::placeholders::_2, ref);
+      std::sort(remaining_segment_indices.begin(), remaining_segment_indices.end(), comp);
```

Comment:
Use of a lambda and std::sort here seems slightly overkill.  Sort runs in O(nlogn), whereas we could just iterate over the available segments doing a typical minimum-search (which would be O(n)).  It also seems (to me) that this code would be simpler and easier to understand.
```
// Find the segment whose first waypoint is closest to the reference position
double min_dist = std::numeric_limits<double>::max()
std::size_t min_index = 0;
for (std::size_t i = 0; i < remaining_segment_indices.size(); ++i)
{
  double dist = (path.at(i).at(0).translation() - ref).norm();
  if (dist < min_dist)
  {
    min_dist = dist;
    min_index = i;
  }
}

segment_order.push_back(min_index);
```

Reply:
Good catch; will change